optimum/onnxruntime/modeling_diffusion.py

```
@@ -0,0 +1,412 @@
+                    CONFIG_NAME,
+                    cls.config_name,
+                }
+            )
+            # Downloads all repo's files matching the allowed patterns
+            model_id = snapshot_download(
+                model_id,
+                cache_dir=cache_dir,
+                local_files_only=local_files_only,
+                use_auth_token=use_auth_token,
+                revision=revision,
+                allow_patterns=allow_patterns,
+                ignore_patterns=["*.msgpack", "*.safetensors", "*.bin"],
+            )
+        new_model_save_dir = Path(model_id)
+        for name in sub_models_names:
+            library_name, library_classes = sub_models_to_load[name]
+            if library_classes is not None:
+                library = importlib.import_module(library_name)
+                class_obj = getattr(library, library_classes)
+                load_method = getattr(class_obj, "from_pretrained")
+                # Check if the module is in a subdirectory
+                if (new_model_save_dir / name).is_dir():
+                    sub_models[name] = load_method(new_model_save_dir / name)
+                else:
+                    sub_models[name] = load_method(new_model_save_dir)
+
+        inference_sessions = cls.load_model(
+            vae_decoder_path=new_model_save_dir / DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER / vae_decoder_file_name,
+            text_encoder_path=new_model_save_dir / DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER / text_encoder_file_name,
+            unet_path=new_model_save_dir / DIFFUSION_MODEL_UNET_SUBFOLDER / unet_file_name,
+            provider=provider,
+            session_options=session_options,
+            provider_options=provider_options,
+        )
+
+        if model_save_dir is None:
+            model_save_dir = new_model_save_dir
+
+        if use_io_binding:
+            logger.warning("OBinding is not yet available, `use_io_binding` set to False.")
```

I would rather raise an error here.